tuxemon/core/states/dialog/__init__.py

```diff
@@ -60,3 +63,29 @@ def next_text(self):
             return text
         except IndexError:
             return None
+
+    def draw_avatar(self):
+        if not self.avatar:
+            return
+
+        # If the prefix is "monster" the second parameter refers to a monster
+        # If this parameter is a number, we're referring to a monster slot in the party
+        # If this parameter is a string, we're referring to a monster name
+        # If there's no prefix, the parameter represents the path to a sprite file
+        params = self.avatar.split(" ")
```

Comment:
im not really a fan of this.  if we are storing items on the dialog class, such as `avatar`, it shouldn't be a raw string that we need to parse.  items stored on objects shouldn't require additional steps to get their meaning.  I think the most clean way to would be storing the "avatar" as a pygame surface, so it is just blit in the correct place.

this logic for decoding the meaning of the "avatar string" belongs in the code where the string is read; it should be in the dialog event action code.  please move it there and set the dialog avatar as a pygame surface.

Reply:
Better yet would be setting the avatar as a pygame sprite, but i'm not going to expect that.  a surface here is fine and it can be cleaned up later and refactored as a sprite.  In general, sprites should be our basic "building block" of graphical elements.  Again, not needed now, but just an FYI

Reply:
TLDR; please move this entire function to the dialog event.  only pass a pygame surface (or pyganim amimation) to the dialog state as the avatar.

Reply:
I see what you mean. Only issue is that there are several dialog events (at least 4): This would add yet more code duplication if done everywhere instead of a common function. Where should such a function be placed though?

Reply:
Fixed in the latest commit: The parameter now takes only one value, which is either a number or a string. When a number it's a monster slot, when a string it refers to a monster by name. Direct file paths can no longer be specified.

The function to pick the avatar was moved to tools.py and only used once when initializing the dialog. The dialog state is now called with the sprite that was fetched, and is only responsible for adding it on startup the same way it adds the text area.